### ISSUE TITLE
[Bug]: 

### ISSUE THREAD

**Issue by Jappie3** [polybar/polybar#2816]
### Checklist

- [X] I have read the appropriate section in the [contributing guidelines](https://github.com/polybar/polybar/blob/master/CONTRIBUTING.md)
- [X] I believe this issue is a problem with polybar itself and not a misconfiguration on my part
- [X] I have searched for other open and closed [issues](https://github.com/polybar/polybar/issues?q=is%3Aissue) that may have already reported this problem
- [X] I have checked the [known issues](https://github.com/polybar/polybar/wiki/Known-Issues) page for this problem.
- [X] I have followed the [debugging guide](https://github.com/polybar/polybar/wiki/Debugging-your-Config) to narrow down the problem to a minimal config.

### Steps to reproduce

Use the minimal config, this is the launch script I use:

```bash
#!/usr/bin/env sh

# Terminate already running bar instances
killall -q polybar
echo "Running instances killed..."

# Wait until the processes have been shut down
while pgrep -x polybar >/dev/null; do sleep 1; done

# Launch main bar & reload on save
polybar -l trace -c ~/.config/polybar/test/config.ini main -r 2>/tmp/log.txt
#polybar -c ~/.config/polybar/test/config.ini main -r &

echo "Bar launched..."
```

### Minimal config

```dosini
[bar/main]
enable-ipc = true
modules-center = Kitty

[module/Kitty]
# Launch Kitty
type = custom/ipc
format = X
format-font = 3
format-background = ${color.davysGrey}

hook-0 = kitty &
;kitty > /dev/null 2>&1 &
;kstart kitty
hook-1 = konsole > /dev/null 2>&1 &
click-left = "#Kitty.hook.0"
click-right = "#Kitty.hook.1"
```

### Polybar log

```text
polybar|notice:  Parsing config file: /home/jasper/.config/polybar/test/config.ini
polybar|trace: config_parser: Parsing /home/jasper/.config/polybar/test/config.ini
polybar|info:  Created legacy ipc fifo at '/tmp/polybar_mqueue.46170'
polybar|info:  Opening ipc socket at '/run/user/1000/polybar/ipc.46170.sock'
polybar|notice:  Listening for IPC messages (PID: 46170)
polybar|info:  Loaded monitor eDP (1920x1080+0+0)
polybar|info:  Configured DPI = 96x96
polybar|info:  Bar geometry: 1920x24+0+0; Borders: 0,0,0,0
polybar|trace: bar: Attach X event sink
polybar|trace: bar: Attach signal receiver
polybar|trace: controller: Setup user-defined modules
polybar|notice:  Loading module 'Kitty' of type 'custom/ipc'
polybar|info:  module/Kitty: Loaded 2 hooks
polybar|error: Invalid value for "module/Kitty.format-background", using default value (reason: The reference ${color.davysGrey} does not exist (no fallback set))
polybar|notice:  Loaded 1 modules
polybar|info:  Starting application
polybar|trace: controller: Main thread id = 1
polybar|info:  Starting module/Kitty
polybar|info:  Entering event loop (thread-id=1)
polybar|trace: module/Kitty: Thread id = 2
polybar|trace: bar: Create renderer
polybar|trace: renderer: Get TrueColor visual
polybar|trace: renderer: Allocate colormap
polybar|trace: renderer: Allocate output window
polybar|trace: renderer: Allocate window pixmaps
polybar|trace: renderer: Allocate graphic contexts
polybar|trace: renderer: Allocate alignment blocks
polybar|trace: renderer: Allocate cairo components
polybar|trace: renderer: Load fonts
polybar|warn:  No fonts specified, using fallback font "fixed"
polybar|notice:  Loaded font "fixed" (name=Noto Sans, offset=0, file=/usr/share/fonts/noto/NotoSans-Regular.ttf)
polybar|info:  Bar window: 0x3000001
polybar|trace: bar: Reconfigure window
polybar|trace: bar: Set window WM_NAME
polybar|trace: bar: Set window _NET_WM_WINDOW_TYPE
polybar|trace: bar: Set window _NET_WM_STATE
polybar|trace: bar: Set window _NET_WM_DESKTOP
polybar|trace: bar: Set window _NET_WM_PID
polybar|trace: bar: Map window
polybar|trace: bar: Draw empty bar
polybar|trace: bar: Setup tray manager
polybar|info:  Disabling tray manager (reason: missing `tray-position`)
polybar|info:  module/Kitty: Rebuilding cache
polybar|trace: bar: Force update
polybar|info:  Redrawing bar window
polybar|trace: renderer: flush(2 geom=9x24+956+0, falloff=0)
polybar|trace: bar: Received expose event
polybar|trace: bar: Received button press: 1 at pos(960, 8)
polybar|trace: Found matching input area
polybar|trace: controller: Queueing input event '#Kitty.hook.0'
polybar|trace: controller: Dequeueing inputdata: '#Kitty.hook.0'
polybar|trace: controller: Processing inputdata: #Kitty.hook.0
polybar|info:  Forwarding action to modules (module: 'Kitty', action: 'hook', data: '0')
polybar|trace: command: Sending SIGTERM to running child process (46173)
polybar|trace: command: Waiting for pid 46173 to finish...
polybar|trace: command: Exited with status 0
polybar|info:  Delivered action to 1 module
polybar|info:  module/Kitty: Rebuilding cache
polybar|trace: bar: Ignoring update (unchanged)
polybar|trace: bar: Received button press: 1 at pos(960, 12)
polybar|trace: Found matching input area
polybar|trace: controller: Queueing input event '#Kitty.hook.0'
polybar|trace: bar: Received button press: 1 at pos(963, 8)
polybar|trace: Found matching input area
polybar|trace: controller: Queueing input event '#Kitty.hook.0'
polybar|trace: controller: Dequeueing inputdata: '#Kitty.hook.0'
polybar|trace: controller: Processing inputdata: #Kitty.hook.0
polybar|info:  Forwarding action to modules (module: 'Kitty', action: 'hook', data: '0')
polybar|trace: command: Sending SIGTERM to running child process (46206)
polybar|trace: command: Waiting for pid 46206 to finish...
polybar|trace: command: Exited with status 0
polybar|info:  Delivered action to 1 module
polybar|trace: controller: Dequeueing inputdata: '#Kitty.hook.0'
polybar|trace: controller: Processing inputdata: #Kitty.hook.0
polybar|info:  Forwarding action to modules (module: 'Kitty', action: 'hook', data: '0')
polybar|trace: command: Sending SIGTERM to running child process (46235)
polybar|trace: command: Waiting for pid 46235 to finish...
polybar|trace: command: Exited with status 0
polybar|info:  Delivered action to 1 module
polybar|info:  module/Kitty: Rebuilding cache
polybar|trace: bar: Ignoring update (unchanged)
polybar|notice:  Received signal(2): Interrupt
polybar|info:  Eventloop finished
polybar|notice:  Termination signal received, shutting down...
polybar|trace: controller: Detach signal receiver
polybar|trace: controller: Stop modules
polybar|info:  module/Kitty: Stopping
polybar|trace: module/Kitty: Release sleep lock
polybar|info:  Deconstruction of module/Kitty took 0 ms.
polybar|trace: module/Kitty: Deconstructing
polybar|trace: ipc: Removing named pipe at: /tmp/polybar_mqueue.46170
polybar|info:  Waiting for spawned processes to end
polybar|info:  Reached end of application...
```

### Expected behavior

Kitty (or any other program) is launched via a click on the ipc hook module and the bar behaves as normal.

### Actual behavior

Polybar opens Kitty and a zombie process is created. The bar freezes completely and will only respond once I close the Kitty window (this also happened with Konsole, those are the only 2 I tested but I believe this to be an issue on Polybar's side).

The zombie process disappears when I close Kitty and after that the inputs I gave to the bar are processed (i.e. changing desktops). I am quite sure that I can work around this by just using a script module, but I don't think this behavior is intended so I decided to make a bug report.

### Window Manager and Version

kwin 5.25.5

### Linux Distribution

Endeavouros

### Polybar version

```text
polybar 3.6.3

Features: +alsa +curl +i3 +mpd +network(libnl) +pulseaudio +xkeyboard
```

### Additional Context / Screenshots

Screenshot of htop showing the zombie process
![image](https://user-images.githubusercontent.com/42720120/189541664-f123e9fe-77bb-47a6-b5e4-9bb921f2b611.png)

**Comment by patrick96** [comment]
Since this is a reduced example, what is your use-case for launching applications using ipc hooks? The purpose of ipc hooks is to run a program, capture its output and print that to the bar, in your example you could simply do `click-left = kitty` (and you don't even need to use the ipc module for that, the text or menu module are better suited for that).

Anyway, this touches on some long standing issues in polybar, module actions can freeze the entire bar (and some other issues with modules running asynchronously). We track that stuff in #1778 and #2337

**Comment by Jappie3** [comment]
I hadn't used IPC hooks before and wanted to see what they could do, then I stumbled upon this issue. Like you said, I can launch programs in a dozen other ways but I decided to report this since the only place where I found someone mention zombie processes with IPC hooks was a [Reddit thread from 3 years ago](https://www.reddit.com/r/Polybar/comments/b8l15e/bar_sometimes_hangs_with_ipc_module/). Seeing as this is apparently related to other stuff you're aware of I'll just close this issue. Thanks for your reply ^^
